File: backend/src/main/java/com/zihler/wiki/domain/values/Body.java
package com.zihler.wiki.domain.values;

import java.util.Objects;

public class Body {
    private String body;

    private Body(String body) {
        this.body = body;
    }

    public static Body from(String body) {
        return new Body(body);
    }

    public static Body empty() {
        return new Body("");
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Body body1 = (Body) o;
        return Objects.equals(body, body1.body);
    }

    @Override
    public int hashCode() {
        return Objects.hash(body);
    }

    @Override
    public String toString() {
        return body;
    }
}

File: frontend/src/app/wiki/domain/body.ts
export class Body {
  constructor(private _body: string) {

  }

  get body(): string {
    return this._body;
  }

  set body(value: string) {
    this._body = value;
  }

  static empty() {
    return this.from('');
  }

  static from(body: string) {
    return new Body(body);
  }
}
